pkg/driver/mount.go

```diff
@@ -230,6 +234,11 @@ func (m *S3Mounter) IsMountPoint(target string) (bool, error) {
 	}
 	for _, mp := range mountPoints {
 		if mp.Path == target {
+			if mp.Device != mountpointDeviceName {
```

Comment:
Can we add a test for this?

Reply:
Added unit-tests with https://github.com/awslabs/mountpoint-s3-csi-driver/pull/258/commits/57b70d260c9b03ef0984e5f672cd74d9c5fee9bc. Wanted to add integration-tests as well, but we'd need to spawn a privileged Pod to make some assertions on the underlying host like checking `/proc/mounts` or something. Not sure if that's worth it